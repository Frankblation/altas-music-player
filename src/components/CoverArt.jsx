import React from "react"
import placeholder from  "../assets/placeholder.svg"

const CoverArt = ({ imageUrl }) => {
  return (
    <div className="w-[400px] h-[400px] rounded-lg overflow-hidden shadow-lg flex items-center justify-center bg-black">
      <img src={ placeholder  || imageUrl } alt="Album Cover" className="max-w-full max-h-full object-cover" />
    </div>
  )
}

export default CoverArt