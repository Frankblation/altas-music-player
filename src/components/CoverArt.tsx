import { render, screen } from "@testing-library/react";
import { expect, test } from "vitest";
import CoverArt from "../components/CoverArt";
import "@testing-library/jest-dom";


test("Props Passed correctly", () => {
  const testCoverUrl = "https://example.com/test-cover.jpg"; // Mock URL

  render(<CoverArt cover={testCoverUrl} />);

  const imgElement = screen.getByRole("img", { name: "Album Cover" });

  // Check if the image has the correct src attribute
  expect(imgElement).toHaveAttribute("src", testCoverUrl);
});