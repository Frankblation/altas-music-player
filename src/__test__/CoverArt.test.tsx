import { render } from "@testing-library/react";
import { expect, test } from "vitest";
import CoverArt from "../components/CoverArt"; // Fix the import if needed

test("CoverArt renders correctly", () => {
  const { container } = render(<CoverArt cover="https://example.com/cover.jpg" />);
  expect(container).toMatchSnapshot();
});