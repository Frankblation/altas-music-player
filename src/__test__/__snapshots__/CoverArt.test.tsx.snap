// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CoverArt renders correctly 1`] = `
<div>
  <div
    class="w-[400px] h-[400px] rounded-3xl overflow-hidden shadow-xlg flex items-center justify-center bg-black"
  >
    <img
      alt="Album Cover"
      class="max-w-full max-h-full object-cover"
      src="https://example.com/cover.jpg"
    />
  </div>
</div>
`;
